### ISSUE TITLE
[local_auth] Support for `macOS` platform

### ISSUE THREAD

**Issue by OutdatedGuy** [flutter/flutter#140685]
### Use case

The [local_auth](https://pub.dev/packages/local_auth) plugin currently supports only android, ios and windows platform. It would be great to have support for macOS as well.

### Proposal

We can copy the implementation of [local_auth_ios](https://pub.dev/packages/local_auth_ios) and make necessary and proprietary changes to support macOS platform.

**Comment by stuartmorgan** [comment]
> We can copy the implementation of [local_auth_ios](https://pub.dev/packages/local_auth_ios)

We don't copy implementations, since that would require maintaining two copies forever; instead we share them. See [path_provider](https://github.com/flutter/packages/tree/main/packages/path_provider/path_provider_foundation), [shared_preferences](https://github.com/flutter/packages/tree/main/packages/shared_preferences/shared_preferences_foundation), and [video_player](https://github.com/flutter/packages/tree/main/packages/video_player/video_player_avfoundation) for some examples.

The first step in adding macOS support would be renaming the existing package iOS. Normally we use the name of the shared implementing framework; unfortunately in this case that's the `LocalAuthentication` framework, which would make the name `local_auth_localauthentication`. Given that I think we'll want to fall back to the less accurate, but more useful in distinguishing the implementation packages, `local_auth_darwin`.

(For anyone interested in doing this, renaming needs to be done in its own PR, without any macOS additions, and then adding macOS would be a follow-up PR once the rename has landed.)

**Comment by stuartmorgan** [comment]
> Normally we use the name of the shared implementing framework; unfortunately in this case that's the `LocalAuthentication` framework, which would make the name `local_auth_localauthentication`. Given that I think we'll want to fall back to the less accurate, but more useful in distinguishing the implementation packages, `local_auth_darwin`.

/cc @jmagman in case you have any alternate thoughts for our latest installment of "Name That Shared iOS/macOS Code!" 🙂.

**Comment by OutdatedGuy** [comment]
> > We can copy the implementation of [local_auth_ios](https://pub.dev/packages/local_auth_ios)
> 
> We don't copy implementations, since that would require maintaining two copies forever; instead we share them. See [path_provider](https://github.com/flutter/packages/tree/main/packages/path_provider/path_provider_foundation), [shared_preferences](https://github.com/flutter/packages/tree/main/packages/shared_preferences/shared_preferences_foundation), and [video_player](https://github.com/flutter/packages/tree/main/packages/video_player/video_player_avfoundation) for some examples.
> 
> The first step in adding macOS support would be renaming the existing package iOS. Normally we use the name of the shared implementing framework; unfortunately in this case that's the `LocalAuthentication` framework, which would make the name `local_auth_localauthentication`. Given that I think we'll want to fall back to the less accurate, but more useful in distinguishing the implementation packages, `local_auth_darwin`.
> 
> (For anyone interested in doing this, renaming needs to be done in its own PR, without any macOS additions, and then adding macOS would be a follow-up PR once the rename has landed.)

Hi, I don't quite understand what you mean. Can you elaborate more on this statement with a simple explanation.

**Comment by stuartmorgan** [comment]
I'm not sure what elaboration you are looking for. Which part isn't clear?

**Comment by OutdatedGuy** [comment]
> The first step in adding macOS support would be renaming the existing iOS package. Normally we use the name of the shared implementing framework; unfortunately in this case that's the `LocalAuthentication` framework, which would make the name `local_auth_localauthentication`. Given that I think we'll want to fall back to the less accurate, but more useful in distinguishing the implementation packages, `local_auth_darwin`.

I didn't understand what you are telling me to do

**Comment by stuartmorgan** [comment]
I'm not telling you to do anything; what you choose to contribute is entirely up to you. I'm describing what we would accept for adding an endorsed macOS implementation of `local_auth`, which is a simple package containing the shared iOS and macOS implementation.

The process for getting from the current state to that state is at least three separate PRs:
1. Rename `local_auth_ios` to something more generic, as described above.
2. Add macOS support to that renamed package.
3. Switch `local_auth` to using the new package instead of `local_auth_ios`.

The relative order of 2 and 3 isn't particularly important, but 1 must be done first.

**Comment by depombo** [comment]
looking forward to this!

**Comment by Rimantovas** [comment]
Any news when it will be merged?
